lib/services/database.go

```diff
@@ -427,6 +429,74 @@ func NewDatabaseFromAzureManagedSQLServer(server *armsql.ManagedInstance) (types
 		})
 }
 
+// NewDatabaseFromAzureMySQLFlexServer creates a database resource from an Azure MySQL Flexible server.
+func NewDatabaseFromAzureMySQLFlexServer(server *armmysqlflexibleservers.Server) (types.Database, error) {
+	if server.Properties == nil {
+		return nil, trace.BadParameter("missing properties")
+	}
+
+	if server.Properties.FullyQualifiedDomainName == nil {
+		return nil, trace.BadParameter("missing FQDN")
+	}
+
+	labels, err := labelsFromAzureMySQLFlexServer(server)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	var description string
+	if replicaRole, ok := labels[labelReplicationRole]; ok {
+		description = fmt.Sprintf("Azure MySQL Flexible server in %v (%v endpoint)",
+			azure.StringVal(server.Location), strings.ToLower(replicaRole))
+	} else {
+		description = fmt.Sprintf("Azure MySQL Flexible server in %v", azure.StringVal(server.Location))
+	}
+
+	return types.NewDatabaseV3(
+		setAzureDBName(types.Metadata{
+			Description: description,
+			Labels:      labels,
+		}, azure.StringVal(server.Name)),
+		types.DatabaseSpecV3{
+			Protocol: defaults.ProtocolMySQL,
+			URI:      fmt.Sprintf("%v:%v", azure.StringVal(server.Properties.FullyQualifiedDomainName), azure.MySQLPort),
+			Azure: types.Azure{
+				Name:       azure.StringVal(server.Name),
+				ResourceID: azure.StringVal(server.ID),
+			},
+		})
+}
+
+// NewDatabaseFromAzurePostgresFlexServer creates a database resource from an Azure PostgreSQL Flexible server.
+func NewDatabaseFromAzurePostgresFlexServer(server *armpostgresqlflexibleservers.Server) (types.Database, error) {
```

Comment:
We should create a ticket for later. Azure requires replicas to have a unique name, so it should be able to discover them unambiguously anyway, but I think it will be better UX if we can point out that the db is a replica in the DB description

Reply:
ticket here: https://github.com/gravitational/teleport/issues/19868